### PR TITLE
Fixed folder upload path encoding (including drag&drop)

### PR DESCRIPTION
**Pull request by Julian1998**
## Description

Removed all unnecessary /'s in the upload path

## Related Issue

https://github.com/owncloud/phoenix/issues/1586

## Motivation and Context

Just a small bubfix

## How Has This Been Tested?

- Upload a folder using the folder upload button / drag & drop
- The upload path shouldn't contain unnecessary /'s

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIUpload</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4375/11

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4375/13

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4375/12

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIUpload</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4376/11

<p>













</p></details>

File: apps/files/src/FileUpload.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ class FileUpload {
     const xhr = new XMLHttpRequest()
 
     // Headers
-    xhr.open(this.type, this.url + encodeURIComponent(this.path), true)
+    this.path = this.path.replace(/(^\/+)/, '')
+    xhr.open(this.type, this.url + this.path, true)
```
Comment (LukasHirt) [inline apps/files/src/FileUpload.js]: `encodeURIComponent` needs to stay pls

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4376/13

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4376/12

<p>









</p></details>

**Comment by Julian1998** [comment]
#1754

**Pull request closed without merging**